feat: Add wasm support for `pages functions build` (#2685)
cloudflare/workers-sdk: .changeset/thirty-hornets-behave.md

---
"wrangler": patch
---

> Currently it is not possible to import `wasm` modules in either Pages
> Functions or Pages Advanced Mode projects.
> 
> This commit caries out work to address the aforementioned issue by
> enabling `wasm` module imports in `wrangler pages functions build`.
> As a result, Pages users can now import their `wasm` modules withing
> their Functions or `_worker.js` files, and `wrangler pages functions build` will correctly bundle everything and output the expected result
> file.

Reviewer: I think Changesets should be less of a 'commit message', and more of a 'release note'.

I'd maybe rephrase to something like:

```suggestion
You can now import Wasm modules in Pages Functions and Pages Functions Advanced Mode (`_worker.js`).

This change specifically enables `wasm` module imports in `wrangler pages functions build`.
As a result, Pages users can now import their `wasm` modules within
their Functions or `_worker.js` files, and `wrangler pages functions build` will correctly bundle everything and output the expected result
file.
```
